atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraServerOrigin.java

```diff
@@ -33,6 +33,6 @@ public static ImmutableMap<CassandraServer, CassandraServerOrigin> mapAllServers
 
     public static ImmutableMap<CassandraServer, CassandraServerOrigin> mapAllServersToOrigin(
             Stream<CassandraServer> servers, CassandraServerOrigin origin) {
-        return servers.collect(ImmutableMap.toImmutableMap(Function.identity(), _v -> origin));
+        return servers.distinct().collect(ImmutableMap.toImmutableMap(Function.identity(), _v -> origin));
```

Comment:
for safety this introduces a constraint on the first argument to `toImmutableMap` being injective! but given urgency, let's get this through